addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -226,10 +226,10 @@ def _prepare_so_line_values(self, order):
         if not float_is_zero(amount_total, precision_rounding=self.currency_id.rounding):
             for line in order.order_line:
                 distrib_dict = line.analytic_distribution or {}
-                for account, distribution in distrib_dict.items():
-                    analytic_distribution[account] = distribution * line.price_total + analytic_distribution.get(account, 0)
-            for account, distribution_amount in analytic_distribution.items():
-                analytic_distribution[account] = distribution_amount/amount_total
+                for account_id, distribution in distrib_dict.items():
+                    analytic_distribution[account_id] = distribution * line.price_total + analytic_distribution.get(account_id, 0)
+            for account_id, distribution_amount in analytic_distribution.items():
+                analytic_distribution[account_id] = distribution_amount / amount_total
```

Comment:
But this doesn't change anything?
@smetl what did you have in mind? Isn't this useless diff?

Reply:
if `account` is in fact an `id`, I agree this diff is useless

Reply:
It is literally taking the keys from one distribution field and putting them in another distribution dict without changing the type...